libs/sdk-ui-charts/src/charts/headline/internal/providers/AbstractProvider.ts

```diff
@@ -0,0 +1,36 @@
+// (C) 2023 GoodData Corporation
+import {
+    ICreateExecutionParams,
+    IHeadlineProvider,
+    IHeadlineTransformationProps,
+} from "../../HeadlineProvider.js";
+import { IExecutionFactory, IPreparedExecution } from "@gooddata/sdk-backend-spi";
+import isEmpty from "lodash/isEmpty.js";
+import { MeasureGroupIdentifier, newDimension } from "@gooddata/sdk-model";
+
+abstract class AbstractProvider implements IHeadlineProvider {
+    public createExecution(
+        executionFactory: IExecutionFactory,
+        params: ICreateExecutionParams,
+    ): IPreparedExecution {
+        const { buckets, filters, executionConfig, dateFormat, sortItems } = params;
+        const execution = executionFactory
+            .forBuckets(buckets, filters)
+            .withExecConfig(executionConfig)
+            .withDimensions(newDimension([MeasureGroupIdentifier]));
+
+        if (dateFormat) {
+            execution.withDateFormat(dateFormat);
+        }
+
+        if (!isEmpty(sortItems)) {
+            execution.withSorting(...sortItems);
+        }
+
+        return execution;
+    }
+
+    public abstract getHeadlineTransformationComponent(): React.ComponentType<IHeadlineTransformationProps>;
+}
+
+export default AbstractProvider;
```

Comment:
Any reason why you going with `default` export in these providers and do not just export the class itself? Personally, I don't like them very much as you need to name the import yourself in target file. I don't see any benefit in them, just inconvenience.

Reply:
Honestly, it's a personal rule and a habit of mine. Whenever I implement a class or a component, I typically use default export to export the main target of that file. My expectation is for each file to ideally serve a single purpose, meaning it should export only one component or class.

This approach is influenced by my previous experience at my former company, where we used ESLint to enforce such rules. You can find more information about this convention here: https://github.com/import-js/eslint-plugin-import/blob/main/docs/rules/prefer-default-export.md. While I acknowledge that this might not be the best solution, it's one of the conventions I've followed.

If our project avoiding default exports would be more appropriate, or if you believe that named exports are better, please let me know. I am open to adopting that approach not only for the code in this pull request but also for my future contributions. Consistency is crucial across our project, and I believe adhering to a unified convention is important for all project members.

Regarding your concern about needing to manually name the import in the target file, I currently use IntelliJ IDE and when I type the name of the default export (in this case, AbstractP), the IDE accurately suggests the intended import: 
 
```typescript
import AbstractProvider from "../AbstractProvider.js";
```
Therefore, I'm not experiencing the issue you mentioned.